### ISSUE TITLE
Multiple use of section label warning for Setex-stype headers in markdown

### ISSUE THREAD

**Issue by gschorcht** [doxygen/doxygen#6779]
doxygen version: 1.8.15
Operating system: Ubuntu 18.04 LTS
Doxyfile: standard

Using Setext-style headers in markdown files throw the warning: `multiple use of section label 'xyz' while adding section, (first occurrence: <file>, line <n>)`.

The problem seems to be that labels are first appended to the sorted dictionary https://github.com/doxygen/doxygen/blob/dc89ac01407c24142698c1374610f2cee1fbf200/src/markdown.cpp#L2303
and are therefore already known when `addSection` is called after that from https://github.com/doxygen/doxygen/blob/dc89ac01407c24142698c1374610f2cee1fbf200/src/commentscan.l#L1780

The problem occures in version 1.8.15 and the current master as well. It wasn't a problem in 1.8.14. I have used the simple test file together with the standard `Doxyfile`.
```md
Main Page                                                           {#mainpage}
=========

Section 1                                                           {#section1}
---------
Test text for section 1 

Section 2                                                           {#section2}
---------
Test text for section 2
```

Maybe the problem is related to #6605, but the problem described there seems not to be related to markdown support. Furthermore, since he claims that the problem didn't occur in 1.8.6, it is not really clear whether his problem report is for 1.8.7 or any newer version. In version 1.8.7 there was a problem that led to the problem described here. Therefore, I open a new bug report since it is in current version an related to markdown support.

**Comment by albert-github** [comment]
Looks like the problem occurs due to the fact that the markdown processing in version 1.8.15 has been moved to a more logical place (otherwise some parts were not processed).

It is not possible to simply remove the test in `addSection` as in that case constructs where twice a section is added with e.g. `\section` command won't give a warning anymore. Look like that removing the relevant code from markdown.cpp should work.

I've just pushed a proposed patch, pull request #6780.

**Comment by albert-github** [comment]
Code (#6780) has been integrated in master on github.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
